### PR TITLE
Terraform files created for all environment

### PR DESCRIPTION
**Pull request by humulla**
I've created the terraform files and the parameter stores for all the environments

### CODE REVIEW

File: terraform/development/main.tf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# 7) ENSURE THIS FILE IS PLACED WITHIN A 'terraform' FOLDER LOCATED AT THE ROOT PROJECT DIRECTORY
+
+provider "aws" {
+  region  = "eu-west-2"
+  version = "~> 2.0"
+}
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+locals {
+  application_name = your application name # The name to use for your application
```
Comment (elcorbs) [inline terraform/development/main.tf:17]: This should be set to a string value of the application name

File: terraform/development/main.tf [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+   parameter_store = "arn:aws:ssm:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:parameter"
+}
+
+terraform {
+  backend "s3" {
+    bucket  = "terraform-state-development-apis"
+    encrypt = true
+    region  = "eu-west-2"
+    key     = services/YOUR API NAME/state #e.g. "services/transactions-api/state"
```
Comment (elcorbs) [inline terraform/development/main.tf:26]: This should be set to a string with like the e.g. comment next to it, but with the API name replaced

File: terraform/development/main.tf [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+data "aws_subnet_ids" "development_private_subnets" {  
+  vpc_id = data.aws_vpc.development_vpc.id  
+  filter {    
+    name   = "tag:Type"    
+    values = ["private"]  
+    }
+}
+
+ data "aws_ssm_parameter" "resident_contact_postgres_db_password" {
```
Comment (elcorbs) [inline terraform/development/main.tf:44]:
```suggestion
 data "aws_ssm_parameter" "resident_contact_postgres_password" {
```

File: terraform/development/main.tf (hunk 2 above)
Comment (Renny4Real) [inline terraform/development/main.tf:26]: Could we rename this to match the API name?

File: terraform/development/main.tf (hunk 1 above)
Comment (Renny4Real) [inline terraform/development/main.tf:17]: Maybe we could add a name for the application?

File: terraform/development/main.tf [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+module "postgres_db_development" {
+  source = "github.com/LBHackney-IT/aws-hackney-common-terraform.git//modules/database/postgres"
+  environment_name = "development"
+  vpc_id = data.aws_vpc.development_vpc.id
+  db_engine = "postgres"
+  db_engine_version = "11.1"
+  db_identifier = "auth-token-generator-dev-db"
+  db_instance_class = "db.t2.micro"
+  db_name = "auth-token-generator_db"
+  db_port  = 5004
```
Comment (elcorbs) [inline terraform/development/main.tf:61]: I believe development port numbers should end in a 1 i.e. ***1

File: terraform/production/main.tf [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+
+data "aws_iam_role" "ec2_container_service_role" {
+  name = "ecsServiceRole"
+}
+
+data "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecsTaskExecutionRole"
+}
```
Comment (elcorbs) [inline terraform/production/main.tf:28]: These can be deleted

File: terraform/production/main.tf [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+# 7) ENSURE THIS FILE IS PLACED WITHIN A 'terraform' FOLDER LOCATED AT THE ROOT PROJECT DIRECTORY
+
+provider "aws" {
+  region  = "eu-west-2"
+  version = "~> 2.0"
+}
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+locals {
+  application_name = your application name # The name to use for your application
```
Comment (elcorbs) [inline terraform/production/main.tf:17]: This should be set to a string value of the application name

File: terraform/production/main.tf [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+data "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecsTaskExecutionRole"
+}
+
+terraform {
+  backend "s3" {
+    bucket  = "terraform-state-production-apis"
+    encrypt = true
+    region  = "eu-west-2"
+    key     = services/YOUR API NAME/state #e.g. "services/transactions-api/state"
```
Comment (elcorbs) [inline terraform/production/main.tf:35]: This should be set to a string with like the e.g. comment next to it, but with the API name replaced

File: terraform/production/main.tf [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+}
+data "aws_subnet_ids" "production_private_subnets" {  
+  vpc_id = data.aws_vpc.production_vpc.id  
+  filter {    
+    name   = "tag:Type"    
+    values = ["private"]  
+    }
+}
+
+ data "aws_ssm_parameter" "resident_contact_postgres_db_password" {
```
Comment (elcorbs) [inline terraform/production/main.tf:53]:
```suggestion
 data "aws_ssm_parameter" "resident_contact_postgres_password" {
```

File: terraform/production/main.tf [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+module "postgres_db_production" {
+  source = "github.com/LBHackney-IT/aws-hackney-common-terraform.git//modules/database/postgres"
+  environment_name = "production"
+  vpc_id = data.aws_vpc.production_vpc.id
+  db_engine = "postgres"
+  db_engine_version = "11.1"
+  db_identifier = "auth-token-generator-dev-db"
+  db_instance_class = "db.t2.micro"
+  db_name = "auth-token-generator_db"
+  db_port  = 5004
```
Comment (elcorbs) [inline terraform/production/main.tf:70]: And production port numbers should end in a 0 i.e. ***0

File: terraform/production/main.tf [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  db_username = data.aws_ssm_parameter.auth_token_generator_postgres_username.value
+  db_password = data.aws_ssm_parameter.auth_token_generator_postgres_db_password.value
+  subnet_ids = data.aws_subnet_ids.production_private_subnets.ids
+  db_allocated_storage = 20
+  maintenance_window ="sun:10:00-sun:10:30"
+  storage_encrypted = false
+  multi_az = false //only true if production deployment
```
Comment (elcorbs) [inline terraform/production/main.tf:77]: Like the comment says, this should be true for production

File: terraform/staging/main.tf [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+# 7) ENSURE THIS FILE IS PLACED WITHIN A 'terraform' FOLDER LOCATED AT THE ROOT PROJECT DIRECTORY
+
+provider "aws" {
+  region  = "eu-west-2"
+  version = "~> 2.0"
+}
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+locals {
+  application_name = your application name # The name to use for your application
```
Comment (elcorbs) [inline terraform/staging/main.tf:17]: This should be set to a string value of the application name

File: terraform/staging/main.tf [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+data "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecsTaskExecutionRole"
+}
+
+terraform {
+  backend "s3" {
+    bucket  = "terraform-state-staging-apis"
+    encrypt = true
+    region  = "eu-west-2"
+    key     = services/YOUR API NAME/state #e.g. "services/transactions-api/state"
```
Comment (elcorbs) [inline terraform/staging/main.tf:35]: This should be set to a string with like the e.g. comment next to it, but with the API name replaced

File: terraform/staging/main.tf [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+
+data "aws_iam_role" "ec2_container_service_role" {
+  name = "ecsServiceRole"
+}
+
+data "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecsTaskExecutionRole"
+}
```
Comment (elcorbs) [inline terraform/staging/main.tf:28]: I think these can be removed

File: terraform/staging/main.tf [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+/*    POSTGRES SET UP    */
+data "aws_vpc" "staging_vpc" {  
+  tags = {    
+    Name = "vpc-staging-apis-staging"  
+    }
+}
+data "aws_subnet_ids" "staging_private_subnets" {  
+  vpc_id = data.aws_vpc.development_vpc.id
```
Comment (elcorbs) [inline terraform/staging/main.tf:46]: I think this should be `staging_vpc`

File: terraform/staging/main.tf [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+data "aws_subnet_ids" "staging_private_subnets" {  
+  vpc_id = data.aws_vpc.development_vpc.id  
+  filter {    
+    name   = "tag:Type"    
+    values = ["private"]  
+    }
+}
+
+ data "aws_ssm_parameter" "resident_contact_postgres_db_password" {
+   name = "/api-auth-token-generator/development/postgres-password"
```
Comment (elcorbs) [inline terraform/staging/main.tf:54]: This is staging

File: terraform/staging/main.tf [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+module "postgres_db_staging" {
+  source = "github.com/LBHackney-IT/aws-hackney-common-terraform.git//modules/database/postgres"
+  environment_name = "staging"
+  vpc_id = data.aws_vpc.staging_vpc.id
+  db_engine = "postgres"
+  db_engine_version = "11.1"
+  db_identifier = "auth-token-generator-dev-db"
+  db_instance_class = "db.t2.micro"
+  db_name = "auth-token-generator_db"
+  db_port  = 5004
```
Comment (elcorbs) [inline terraform/staging/main.tf]: Staging port numbers should end in 2

File: terraform/development/main.tf [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+  application_name = "auth token generator api"
+   parameter_store = "arn:aws:ssm:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:parameter"
+}
+
+terraform {
+  backend "s3" {
+    bucket  = "terraform-state-development-apis"
+    encrypt = true
+    region  = "eu-west-2"
+    key     = services/auth-token-generator-api/state #e.g. "services/transactions-api/state"
```
Comment (elcorbs) [inline terraform/development/main.tf:26]: This should be a string

File: terraform/production/main.tf [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+  application_name = "auth token generator api"
+   parameter_store = "arn:aws:ssm:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:parameter"
+}
+
+terraform {
+  backend "s3" {
+    bucket  = "terraform-state-production-apis"
+    encrypt = true
+    region  = "eu-west-2"
+    key     = services/auth-token-generator-api/state #e.g. "services/transactions-api/state"
```
Comment (elcorbs) [inline terraform/production/main.tf:26]: This should be a string

File: terraform/production/main.tf [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+   name = "/api-auth-token-generator/production/postgres-username"
+ }
+
+module "postgres_db_production" {
+  source = "github.com/LBHackney-IT/aws-hackney-common-terraform.git//modules/database/postgres"
+  environment_name = "production"
+  vpc_id = data.aws_vpc.production_vpc.id
+  db_engine = "postgres"
+  db_engine_version = "11.1"
+  db_identifier = "auth-token-generator-dev-db"
```
Comment (elcorbs) [inline terraform/production/main.tf:58]: Should this name reflect that it's in production?

File: terraform/production/main.tf [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+  db_instance_class = "db.t2.micro"
+  db_name = "auth-token-generator_db"
+  db_port  = 5102
+  db_username = data.aws_ssm_parameter.auth_token_generator_postgres_username.value
+  db_password = data.aws_ssm_parameter.auth_token_generator_postgres_password.value
+  subnet_ids = data.aws_subnet_ids.production_private_subnets.ids
+  db_allocated_storage = 20
+  maintenance_window ="sun:10:00-sun:10:30"
+  storage_encrypted = false
+  multi_az = truen
```
Comment (elcorbs) [inline terraform/production/main.tf:68]: Typo here

File: terraform/staging/main.tf [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+  application_name = "auth token generator api"
+   parameter_store = "arn:aws:ssm:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:parameter"
+}
+
+terraform {
+  backend "s3" {
+    bucket  = "terraform-state-staging-apis"
+    encrypt = true
+    region  = "eu-west-2"
+    key     = services/auth-token-generator-api/state #e.g. "services/transactions-api/state"
```
Comment (elcorbs) [inline terraform/staging/main.tf:26]: This should be a string

File: terraform/staging/main.tf [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+   name = "/api-auth-token-generator/staging/postgres-username"
+ }
+
+module "postgres_db_staging" {
+  source = "github.com/LBHackney-IT/aws-hackney-common-terraform.git//modules/database/postgres"
+  environment_name = "staging"
+  vpc_id = data.aws_vpc.staging_vpc.id
+  db_engine = "postgres"
+  db_engine_version = "11.1"
+  db_identifier = "auth-token-generator-dev-db"
```
Comment (elcorbs) [inline terraform/staging/main.tf:58]: Again, should mention staging instead of dev here

**Pull request merged**